osu.Game/Graphics/UserInterface/OsuNumberBox.cs

```diff
@@ -1,17 +1,17 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using osu.Framework.Input;
+
 namespace osu.Game.Graphics.UserInterface
 {
     public partial class OsuNumberBox : OsuTextBox
     {
-        protected override bool AllowIme => false;
-
         public OsuNumberBox()
         {
+            InputProperties = new TextInputProperties(TextInputType.Number, false);
+
             SelectAllOnFocus = true;
         }
-
-        protected override bool CanAddCharacter(char character) => char.IsAsciiDigit(character);
```

Comment:
can be done because of https://github.com/ppy/osu-framework/blob/416718c231aaeb31cd2b9c64e0388688eef2fd36/osu.Framework/Graphics/UserInterface/TextBox.cs#L98